RepositoryCompilerTests/Unit/RepositoryCompilerAMQPTests.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using RepositoryCompiler.Communication;
+using RepositoryCompilerTests.DataFactories;
+using Shouldly;
+using Xunit;
+
+namespace RepositoryCompilerTests.Unit
+{
+    public class RepositoryCompilerAMQPTests
+    {
+        private readonly MetricsReportFactory _metricsReportFactory = new MetricsReportFactory();
+
+        [Fact]
+        public void Produce_Metrics_Report_Message()
+        {
+            RepositoryCompilerMessageProducer  producer = new RepositoryCompilerMessageProducer();
+            CaDETClassDTO reportMessage = _metricsReportFactory.CreateMockupMetricsReportMessage();
+            producer.CreateNewMetricsReport(reportMessage);
```

Comment:
Since this is not an actual test, we will need to remove it before completing the PR.